src/components/auth/AuthButton.js

```diff
@@ -63,7 +67,20 @@ export default class AuthButton extends React.Component {
     }
 
     return (
-      <Button type="primary" onClick={() => startAuthenticationFlow(this.props.location.pathname)}>
+      <Button
+        type="primary"
+        loading={this.props.authStarted}
+        onClick={() => {
+          this.props.setAuthStarted(true);
+          startAuthenticationFlow(this.props.location.pathname);
```

Comment:
Please advice; if this was regular `setState` I would have done:
```javascript
this.setState({authStarted: true}, () => {
  startAuthenticationFlow(this.props.location.pathname);
  ...
})
```
that way the renderer gets a chance to re-render with this state change fully before doing anything else. 

How do you do that with Redux? Some quick googling told me that you don't but you can make the action return a `Promise` and then you'd be able to...:
```javascript
this.props.setAuthStarted(true).then(() => {
  ...
})
```

Reply:
What you should do is move this into the action. see my note about that.